src/current/v24.3/ldap-authentication.md

```diff
@@ -0,0 +1,149 @@
+
+~~~ sql
+SET CLUSTER SETTING server.redact_sensitive_settings.enabled = 'true';
+~~~
+
+### Step 2: Configure Host-Based Authentication (HBA)
+
+Set the `server.host_based_authentication.configuration` cluster setting to enable LDAP authentication.
+
+The format for HBA is `TYPE  DATABASE  USER  ADDRESS  METHOD  [OPTIONS]`. For LDAP, the method should be set to `ldap` and the option parameters that follow will be LDAP specific:
+
+- `ldapserver`: LDAP server hostname
+- `ldapport`: LDAP server port (typically 636 for LDAPS)
+- `ldapbasedn`: Base DN for user searches
+- `ldapbinddn`: Service account DN for directory searches
+- `ldapbindpasswd`: Service account password
+- `ldapsearchattribute`: Attribute to match against SQL usernames
+- `ldapsearchfilter`: LDAP filter to restrict valid users
+
+For example:
+
+~~~ sql
+SET CLUSTER SETTING server.host_based_authentication.configuration = '
+host    all    all    all    ldap    ldapserver=ldap.example.com 
+    ldapport=636 
+    "ldapbasedn=ou=users,dc=example,dc=com" 
+    "ldapbinddn=cn=readonly,dc=example,dc=com" 
+    ldapbindpasswd=readonly_password 
+    ldapsearchattribute=uid 
+    "ldapsearchfilter=(memberof=cn=cockroachdb_users,ou=groups,dc=example,dc=com)"';
+~~~
+
+If you also intend to [configure LDAP Authorization]({% link v24.3/ldap-authorization.md %}#configuration), you will need to include an additional LDAP parameter, `ldapgrouplistfilter`.
+
+For more information on the HBA fields and HBA in CockroachDB, generally, refer to [SQL Authentication configuration](https://www.cockroachlabs.com/docs/stable/security-reference/authentication#authentication-configuration).
+
+### Step 3: Configure TLS (Optional)
+
+If using LDAPS with a certificate signed by a custom Certificate Authority (CA) that is not in the system's trusted CA store, you'll need to configure the CA certificate. This step is only necessary when using certificates signed by your organization's private CA or other untrusted CA.
+
+Set the custom CA certificate:
+
+~~~ sql
+SET CLUSTER SETTING server.ldap_authentication.domain_ca = '<PEM_ENCODED_CA_CERT>';
+~~~
+
+Configure a client certificate for mTLS if required:
+
+~~~ sql
+SET CLUSTER SETTING server.ldap_authentication.client.tls_certificate = '<PEM_ENCODED_CERT>';
+SET CLUSTER SETTING server.ldap_authentication.client.tls_key = '<PEM_ENCODED_KEY>';
+~~~
+
+### Step 4: Sync Database Users
+
+Before LDAP authentication can be used for a user, the username must be created directly in CockroachDB. You will, therefore, need to establish an automated method for keeping users in sync with the directory server, creating and dropping them as needed.
+
+For Azure AD deployments, the username typically corresponds to the `sAMAccountName` field from AD's person object, which is specified in the HBA configuration using `ldapsearchattribute=sAMAccountName`.
```

Comment:
Do we want to give any other examples besides Azure AD? Want to use the new name for AAD?

Reply:
We should call it Microsoft Entra ID (formerly called Azure AD)
https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id

Reply:
I looked further into this, and it looks like:

- For Self-Managed Active Directory: 
  - It's the `user` object (not `person` object), `sAMAccountName` field. These are short and typically compatible with our username requirements.
  
- For Entra ID (formerly Azure AD):
  - The `user` object instead has a `UserPricinpalName` field, considered to be the replacement for `sAMAccountName`.
  - This can't be used as-is because it has an `@` symbol (`<local_name>@<domain>`).
  - As an alternative, a customer can likely use `mailNickname`. Here's how it is set for a user:
    - When created through Azure Portal: `mailNickname` is automatically generated based on the user's entered username (the local part of what will become their UPN).
    - When synced from self-managed AD: The `mailNickname` is typically derived from the on-premises attributes. It often matches the `sAMAccountName` from self-managed AD.
    - When created programmatically (Graph API, PowerShell):`mailNickname` is a required attribute, but no automatic matching happens; the customer is responsible for setting it.
  - Or, I suppose the customer could set up a new field just for specifying a CRDB username by splitting or otherwise transforming the UPN.
    
I haven't fully verified all of these details so I'm happy to leave it at the Active Directory `sAMAccountName` example, for now, with some slight refinements, unless @souravcrl or @dikshant would prefer to confirm/include an Entra ID recommendation here.

Reply:
Thank you. Despite what I said on Slack, if you think it's safe to assume it's there for all/most Entre ID instances, we might as well put it back in. Or we can wait if more verification would be helpful.